src/pages/createConfig/step2AddConfig/step2AddConfig.tsx

```diff
@@ -55,30 +64,58 @@ export const Step2AddConfig: React.FC<Step2AddConfigProps> = ({ onDataChange, on
         },
       ],
     };
+    return options;
   }, [schema, isSuccess, schemaId]);
 
-  useEffect(() => {
-    const shouldRegister = monaco?.languages.json.jsonDefaults.diagnosticsOptions.schemas?.length === 0;
-    if (shouldRegister) {
-      registerRefSnippet(monaco);
-    }
-  }, [diagnosticOptions, monaco]);
-
   useEffect(() => {
     if (!monaco) {
       return;
     }
     monaco.languages.json.jsonDefaults.setDiagnosticsOptions(diagnosticOptions);
+    const snippetProvider = registerRefSnippet(monaco);
+    const hoverProvider = registerRefHoverProvider(monaco);
+
+    return () => {
+      snippetProvider.dispose();
+      hoverProvider.dispose();
+    };
   }, [diagnosticOptions, monaco]);
 
   const handleChange = async (value: string | undefined) => {
+    let jsonDereferenced = value ?? '{}';
+    onJsonStringChange(value);
+
     try {
-      onJsonStringChange(value);
-      const json = JSON.parse(value ?? '');
-      const isValid = await validate(json);
+      if (value !== undefined && isRef(value)) {
+        setIsLoading(true);
 
+        jsonDereferenced = await dereference(value);
+        setNetworkError(undefined);
+      }
+    } catch (error) {
+      if (error instanceof ApiError) {
+        setNetworkError(error.message);
+      } else {
+        setNetworkError(undefined);
+      }
+    } finally {
+      setIsLoading(false);
+    }
+
```

Comment:
if there is a network error, fail the validation

Reply:
done